airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -0,0 +1,208 @@
+        next_page_token: Mapping[str, Any] = None,
+    ) -> str:
+        class_name = self.__class__.__name__
+        return f"{class_name[0].lower()}{class_name[1:]}"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"api_token": self._api_token}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field) or []
+        yield from records
+
+
+class PaginatedPipedriveStream(PipedriveStream):
+    limit = 50
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        pagination_data = response.json().get("additional_data", {}).get("pagination", {})
+        if pagination_data.get("more_items_in_collection"):
+            start = pagination_data.get("start", 0) + self.limit
+            return {"start": start}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        next_page_token = next_page_token or {}
+        params.update({"limit": self.limit, **next_page_token})
+        return params
+
+
+class IncrementalPipedriveStream(PaginatedPipedriveStream, ABC):
+    cursor_field = "update_time"
+
+    def __init__(self, replication_start_date: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self._replication_start_date = replication_start_date
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    @property
+    @abstractmethod
+    def path_param(self):
+        pass
```

Comment:
```suggestion
        return self.name[:-1]
```
This might be helpful